FreeRTOS-Plus/Demo/AWS/Ota_Windows_Simulator/Ota_Over_Http_Demo/DemoTasks/OtaOverHttpDemoExample.c

```diff
@@ -0,0 +1,2558 @@
+    }
+    else
+    {
+        LogError( ( "Error: No OTA data buffers available.\r\n" ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+static void prvMqttDefaultCallback( void * pvIncomingPublishCallbackContext,
+                                    MQTTPublishInfo_t * pxPublishInfo )
+{
+    bool isMatch = false;
+
+    ( void ) MQTT_MatchTopic( pxPublishInfo->pTopicName,
+                              pxPublishInfo->topicNameLength,
+                              OTA_JOB_ACCEPTED_RESPONSE_TOPIC_FILTER,
+                              OTA_JOB_ACCEPTED_RESPONSE_TOPIC_FILTER_LENGTH,
+                              &isMatch );
+
+    if( isMatch == true )
+    {
+        prvMqttJobCallback( pvIncomingPublishCallbackContext, pxPublishInfo );
+    }
+}
+
+/*-----------------------------------------------------------*/
+static void prvMqttDataCallback( void * pvIncomingPublishCallbackContext,
+                                 MQTTPublishInfo_t * pxPublishInfo )
+{
+    OtaEventData_t * pxData;
+    OtaEventMsg_t eventMsg = { 0 };
+
+    configASSERT( pxPublishInfo != NULL );
+    ( void ) pvIncomingPublishCallbackContext;
+
+    LogInfo( ( "Received data message callback, size %zu.\n\n", pxPublishInfo->payloadLength ) );
+
+    pxData = prvOTAEventBufferGet();
+
+    if(pxData != NULL )
+    {
+        memcpy(pxData->data, pxPublishInfo->pPayload, pxPublishInfo->payloadLength );
+        pxData->dataLength = pxPublishInfo->payloadLength;
+        eventMsg.eventId = OtaAgentEventReceivedFileBlock;
+        eventMsg.pEventData = pxData;
+
+        /* Send job document received event. */
+        OTA_SignalEvent( &eventMsg );
+    }
+    else
+    {
+        LogError( ( "Error: No OTA data buffers available.\r\n" ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvProcessIncomingData( void * pxSubscriptionContext,
+                                    MQTTPublishInfo_t * pPublishInfo )
+{
+    configASSERT( pPublishInfo != NULL );
+
+    ( void ) pxSubscriptionContext;
+
+    OtaEventData_t * pData;
+    OtaEventMsg_t eventMsg = { 0 };
+
+    LogDebug( ( "Received OTA image block, size %d.\n\n", pPublishInfo->payloadLength ) );
+
+    configASSERT( pPublishInfo->payloadLength <= OTA_DATA_BLOCK_SIZE );
+
+    pData = prvOTAEventBufferGet();
+
+    if( pData != NULL )
+    {
+        memcpy( pData->data, pPublishInfo->pPayload, pPublishInfo->payloadLength );
+        pData->dataLength = pPublishInfo->payloadLength;
+        eventMsg.eventId = OtaAgentEventReceivedFileBlock;
+        eventMsg.pEventData = pData;
+
+        /* Send job document received event. */
+        OTA_SignalEvent( &eventMsg );
+    }
+    else
+    {
+        LogError( ( "Error: No OTA data buffers available.\r\n" ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvProcessIncomingJobMessage( void * pxSubscriptionContext,
```

Comment:
I believe this was merged with `prvMqttJobCallback`. Please remove this definition and the above declaration.

Reply:
fixed in d97ddb7c4611eb1ab80e6f77956319ef4f97c54f